### PR TITLE
Support Wallet Connect

### PR DESCRIPTION
**Pull request by nwingt**

### CODE REVIEW

File: app/models/wallet-connect-client/wallet-connect-client.ts [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+          case "keplr_sign_amino_wallet_connect_v1":
+            try {
+              result = yield self.env.authCoreAPI.signAmino(
+                payload.params[2],
+                payload.params[1]
+              )
+            } catch (error) {
+            }
+            break
+
```
Comment (williamchong) [inline app/models/wallet-connect-client/wallet-connect-client.ts:214]: Please wrap the cases in `{}`s, and comment what do the `params` mean in each case if possible

File: app/models/wallet-connect-client/wallet-connect-client.ts [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+            break
+  
+          default:
+            break
+        }
+      } catch (error) {
+        logError(error)
+      }
+
+      console.log("handleCallRequestApproval", payload, result)
```
Comment (williamchong) [inline app/models/wallet-connect-client/wallet-connect-client.ts:222]: Please remove console.log

**Review by williamchong — CHANGES_REQUESTED** [review]
We did not use any bridge, or hook into custom walletconnect protocol like `keplr://` right?

File: app/models/wallet-connect-client/wallet-connect-client.ts [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+  .actions(self => ({
+    handleNewSessionRequest(
+      error: Error | null,
+      payload: any | null,
+    ) {
+      if (error) {
+        throw error
+      }
+
+      const { peerId, peerMeta } = payload.params[0]
```
[the remark this answers is not in the document]
Comment (nwingt) [reply app/models/wallet-connect-client/wallet-connect-client.ts:93]: Params are returned in an array and we just need the first param

File: app/screens/wallet-connect-request-screen/wallet-connect-request-screen.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+const HeaderText = styled(Text)`
+  color: ${({ theme }) => theme.color.text.feature.highlight.primary};
+  font-size: ${({ theme }) => theme.text.size.lg};
+  font-weight: 500;
+  text-align: center;
+`
+
+export interface WalletConnectRequestScreenProps extends NavigationStackScreenProps<{
+  peerId: string,
+  peerMeta: any;
```
[the remark this answers is not in the document]
Comment (nwingt) [reply app/screens/wallet-connect-request-screen/wallet-connect-request-screen.tsx:27]: Reserve for showing the content of peerMeta in UI in the future

**Pull request merged**